### PR TITLE
Added ROI selection

### PR DESCRIPTION
**Pull request by PhilipYarra**
These changes add an -r argument to the CLI, allowing the user to specify a regex for ROIs that should be excluded from export. This can be useful to filter out large ROIs (such as RadCalc Bone) or those that are irrelevant outside of the planning system (typically _obj and _sm for us). 

This is my first go at contributing to the project, so I'm aware there may be things that need fixing before it's accepted. Please just let me know if you find any issues.

### CODE REVIEW

File: lib/pymedphys/_experimental/pinnacle/pinnacle.py
Comment (SimonBiggs) [inline lib/pymedphys/_experimental/pinnacle/pinnacle.py:190]: What if this instead took a list of regex filters, and then on the CLI allow it to be passed multiple times to provide multiple ignore strings?

**Review by SimonBiggs — COMMENTED** [review]
Hi @PhilipYarra,

This looks great. And thank you for beginning your first contribution 🥳.

Something extra that will be needed for this PR though will be some tests on example data. Have the test exercise the CLI that you've added and make sure it does indeed ignore/include the ROIs you expect it to.

I'll let @pchlap take the lead on reviewing this one though as he has responsibility for this region of PyMedPhys.

@PhilipYarra if you need any help/examples on how to write the test that I described above shout out 🙂

File: lib/pymedphys/_experimental/pinnacle/pinnacle.py
Comment (PhilipYarra) [reply lib/pymedphys/_experimental/pinnacle/pinnacle.py:190]: Hi Simon, yeah, that's a possibility... I've been running it with -r  '.*_obj|.*_ring|.*_sm|^sm$|.*_olap|normal_tissue|Skin|RadCalc' but could use it multiple times, with the presumed behaviour of ORing them all together. You'd lose some of the expressive power of the regex, but that's not a problem for my use case. 

Personally... I like big regexes, and I cannot lie :-)

I'll add some tests. Will probably need Phil Chlap's help on doing that... next week.

Comment (pchlap) [reply lib/pymedphys/_experimental/pinnacle/pinnacle.py:190]: Hi @PhilipYarra 
Thanks a lot for this submission! At first glance this is looking good! As @SimonBiggs mentioned we can have a look at adding a test for this. I'd be happy to help you out with that a bit later in the week.

I agree with Simon that it might be easier for users to allow them to specify -r multiple times since many users might not be familiar with regex's. But I think we should be able to get the best of both worlds by still supporting regex's within the -r strings...

I'll get back to you on Thurs or so with a more detailed code review as well as a start on some unit tests.

Thanks
Phil

**Review by pchlap — COMMENTED** [review]
Firstly thanks again @PhilipYarra for this submission and apologies for taking a little while to get to this.

Great stuff, really nice feature and this sized contribution is really easy to review and wrap up. I've added a few minor comments and if you wouldn't mind adding the functionality to allow multiple skip ROI arguments that would be awesome since it could really make life easier for many users.

The last thing we'll need to add is a test. I've had a go at putting one together:
```
@pytest.mark.slow
def test_pinnacle_cli_skip_roi(data):

    output_path = tempfile.mkdtemp()

    skip_roi_name = "Target"

    for pinn_dir in data.joinpath("Pt1").joinpath("Pinnacle").iterdir():

        command = (
            [str(pmp_test_utils.get_executable_even_when_embedded()), "-m"]
            + "pymedphys experimental pinnacle export".split()
            + [
                "-o",
                output_path,
                "-m",
                "RTSTRUCT",
                "-r",
                skip_roi_name,
                "-t",
                "Trial_1",
                pinn_dir.as_posix(),
            ]
        )

        subprocess.check_call(command)

    # Check that the ROI excluded is not in the resulting RTStruct
    rts_dcm = os.listdir(output_path)[0]
    ds = pydicom.read_file(os.path.join(output_path, rts_dcm))
    for roi in ds.StructureSetROISequence:
        assert not roi.ROIName == skip_roi_name
```

I can't (at least not easily) add it to your pull request because you are merging from a forked repo. So if you wouldn't mind placing that in `lib/pymedphys/tests/experimental/pinnacle/test_pinnacle_cli.py` to see how it goes? You will also need to add the import: `from pymedphys._imports import pydicom`

There seemed to be a pylint test failing but I couldn't easily tell what the issue with that was. If it continues after your final changes we can ask Simon to have a look at that.

Let me know if you have any questions or need a hand with any of that! :)

**Comment by pchlap** [comment]
Hi @PhilipYarra 
Just wondering how you are going with the comments I left? Let me know if you need a hand with any of this.

Cheers
Phil

**Comment by PhilipYarra** [comment]
Hi Phil,

I haven't had time recently to even look at this. I hope to get some time
in the near future to look at it again.

Cheers
Philip.

**Comment by SimonBiggs** [comment]
Hi @pchlap and @PhilipYarra would you both say that this PR as it stands right now overall improves the code health of PyMedPhys?

See the following comment within [Google's standard of code review](https://google.github.io/eng-practices/review/reviewer/standard.html):

> In general, reviewers should favor approving a [pull request] once it is in a state where it definitely improves the overall code health of the system being worked on, even if the [pull request] isn’t perfect.

Were all of the critical comments addressed? If this PR hangs around too much longer it is going to be in danger of being closed and not merged at all...

**Comment by pchlap** [comment]
Hi @SimonBiggs, reading through the history I believe the only thing really missing was a test for this change. Since I had actually sketched one out in one of my previous comments I will revisit that and just see if I can get that added to this pull request. That will also let me double check that this PR id fine to merge. I'll get back to you by the end of the week with how to proceed here. Thanks.

**Comment by PhilipYarra** [comment]
Thanks Phil.

I'm no longer involved with this project, so my need for the feature is
past, but happy to see it incorporated into the project if others will find
it useful.

Cheers
Philip.

**Comment by pchlap** [comment]
@PhilipYarra just FYI I have made a few adaptations to this and pushed the changes back to this PR. I reverted some changes to the rtplan.py file since they weren't working on my end and were out of the scope of this PR. I believe you aren't using this code anymore anyway but just something to be aware of if you (or any of your old colleagues) come back to this code.

@SimonBiggs looks like the pre-commit pipeline is failing. Without looking into it too closely could this be because this is coming from a forked repo? Perhaps we can discuss this on Thursday.

But other than that I would be happy to see this PR merged. It may well prove useful to a future user and won't have any other impact on the functionality of the tool.  I've added a test and all other tests are passing. Won't merge it just yet until we meet on Thursday.

**Comment by PhilipYarra** [comment]
Thanks Phil, nice to have it available. Who knows if I might need it again
at some point.

**Comment by SimonBiggs** [comment]
> looks like the pre-commit pipeline is failing

Yeah the @pymedphys-bot account needs to have push access to the repo for the pre-commit action to succeed.

File: lib/pymedphys/tests/experimental/pinnacle/test_pinnacle_cli.py
Comment (pchlap) [reply lib/pymedphys/tests/experimental/pinnacle/test_pinnacle_cli.py]: No worries, done. Remind me to pick your brain about these imports tomorrow.

Comment (SimonBiggs) [reply lib/pymedphys/tests/experimental/pinnacle/test_pinnacle_cli.py]: > Remind me to pick your brain about these imports tomorrow.

Yup, before then though I recommend doing a little test to see how they work. Uninstall pydicom, then within a python shell run:

```
from pymedphys._imports import pydicom, numpy

np.linspace(0, 1)  # do something with numpy
sequence = pydicom.Sequence()  # do something with pydicom
```

And you'll see that it no longer errors at import time (can still do the numpy things), it only errors at "use time". Essentially it's an implementation of lazy imports. The import only occurs on first use. Designed so that pymedphys can be installed with minimal or no dependencies if so desired.

File: lib/pymedphys/tests/experimental/pinnacle/test_pinnacle_cli.py
Comment (SimonBiggs) [inline lib/pymedphys/tests/experimental/pinnacle/test_pinnacle_cli.py:145]:
```suggestion
@pytest.mark.slow
@pytest.mark.pydicom
```

**Review by SimonBiggs — APPROVED** [review]
It might be worth also marking the test with pydicom so that it runs within pydicom's testing suite. If you're happy to approve this Phil I am happy for it to merge, so I'll leave my approval here indicating that.

**Comment by SimonBiggs** [comment]
Hi Phil,

I'm happy to have this PR merge if you'd like it to?

Cheers,
Simon

**Pull request merged**
